content/modules/rest/pages/entities-api.adoc

```diff
@@ -0,0 +1,724 @@
+
+{
+  "customerName": "Shelby Robinson"
+}
+----
+
+==== Collection Parameters
+
+It is also possible to define a parameter as a collection type. In this case the query definition should contain a `[]` indicator after the Java type.
+
+[source,xml]
+.rest-queries.xml
+----
+<?xml version="1.0"?>
+<queries xmlns="http://jmix.io/schema/rest/queries">
+    <query name="ordersByIds" entity="rstex11_Order" fetchPlan="order-with-details">
+        <jpql><![CDATA[select e from rstex11_Order e where e.id in :ids]]></jpql>
+        <params>
+            <param name="ids" type="java.util.UUID[]"/> // <1>
+        </params>
+    </query>
+</queries>
+----
+<1> the `ids` parameter is marked as collection of `UUID` type
+
+When this parameter is used in a query, the corresponding  IDs have to be provided as a JSON array.
+
+[source, http request]
+.Query API Collection Parameters Request
+----
+POST http://localhost:8080/rest/queries/rstex11_Order/ordersByIds
+
+{
+  "ids": [
+    "41aae331-b46b-85ee-b0bc-2de8cbf1ab86",
+    "21021f78-edac-224b-e6f8-6e71e02a0f0d"
+  ]
+}
+----
+
+== Create Entities
+
+The Entities API allows creating entities through the Create Entity endpoint. An entity can be created via performing a `POST` request against the endpoint: `/entities/:entityName`.
+
+The request body contains a JSON object with the attributes of the entity.
+
+NOTE: Attributes from xref:data-model:entities.adoc#traits[entity traits] like `id` or `createdBy` should not be part of the request. Those attributes are automatically added by Jmix while storing the entity.
+
+When the entity is created successfully, the HTTP response status code `201 - Created` is returned. By default, a JSON metadata representation of the entity is returned mainly containing the newly created `id` attribute for further reference.
+
+[source, http request]
+.Request
+----
+POST http://localhost:8080/rest/entities/rstex11_Customer
+
+{
+  name: "Randall Bishop"
+}
+----
+
+
+[source, json]
+.Response: 201 - Created
+----
+{
+  "_entityName": "rstex11_Customer",
+  "_instanceName": "Randall Bishop",
+  "id": "78e7996d-8b69-6526-8e9f-16262a1c4113"
+}
+----
+
+Alternatively it is possible to define which attributes should be returned once the entity is created. This can be achieved through the URL query parameter `responseFetchPlan`. E.g. `/entities/rstex11_Order?responseFetchPlan=order-with-details` would return the complete order with all its details containing order lines, customer references etc.
+
+TIP: The HTTP response Header `Location` indicates also the URL to the newly created entity instance for further operations (like fetching, updating or deleting).
+
+=== Entity Validation
+
+When an entity is created or updated by default the regular entity validation is active and enforced. This means that invalid input (according to the entity validation annotations) from the API is rejected with the HTTP response status code `400 - Bad Request`.
```

Comment:
"When an entity is created or updated, the regular entity validation is active and enforced by default."